app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -122,7 +122,7 @@ def detect_matching_host(allowed_hosts, request_hosts)
         rex = /\A#{Regexp.escape(request).sub('\\*', '.*')}\Z/
         return allowed if allowed =~ rex
       else
-        return allowed if allowed == request
+        return allowed if (allowed == request) || (allowed == "[#{request}]")
```

Comment:
Should we use `IPAddr` to try to parse it perhaps?
```
[16] pry(main)> IPAddr.new('[::ffff:127.0.0.1]').native().to_s
=> "127.0.0.1"
```

Reply:
Do you mean comparing IPAddr objects instead? Looks cleaner, yeah.

Reply:
Yes. It should probably be a fallback with a try/except. Since IPAddr does support CIDR notation, that could be useful as well to support rather than regexes.